src/ui/coverage/coverageInfo.ts

```diff
@@ -0,0 +1,283 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+import * as path from 'path';
+
+import * as vscode from 'vscode';
+
+import Config from './config';
+import GlobalConfig from '../../globalConfig';
+import { getKaniPath } from '../../model/kaniRunner';
+import { CommandArgs, getRootDir, splitCommand } from '../../utils';
+
+const { execFile } = require('child_process');
+
+// Interface for parsing Kani's output and storing as an object
+interface CoverageEntry {
+	filePath: string;
+	lineNumber: number;
+	coverageStatus: string;
+}
+
+// Interface for storing the coverage status for each line of a document.
+export interface CoverageLines {
+    full: vscode.Range[];
+    partial: vscode.Range[];
+    none: vscode.Range[];
+}
+
+enum CoverageStatus {
+	Full = "FULL",
+	Partial = "PARTIAL",
+	None = "NONE"
+}
+
+// Callback function for the coverage code lens action
+export async function runCodeCoverageAction(renderer: CoverageRenderer, functionName: string): Promise<void> {
+	const globalConfig = GlobalConfig.getInstance();
+	const kaniBinaryPath = globalConfig.getFilePath();
+
+	const activeEditor = vscode.window.activeTextEditor;
+	const currentFileUri = activeEditor?.document.uri.fsPath;
+
+	const playbackCommand: string = `${kaniBinaryPath} ${currentFileUri} --coverage -Z line-coverage --harness ${functionName}`;
+	const processOutput = await runCoverageCommand(playbackCommand, functionName);
+
+	if(processOutput.statusCode == 0) {
+		const coverageOutputArray = processOutput.result;
+
+		const formatted = parseCoverageFormatted(coverageOutputArray);
+		globalConfig.setCoverage(formatted);
+
+		// log global cached map
+		console.log(formatted);
+
+		renderer.renderInterface(vscode.window.visibleTextEditors, formatted);
+	}
+}
+
+/**
+ * Run the visualize command to generate the report, parse the output and return the result
+ *
+ * @param command - the cargo kani | kani command to run --visualize
+ * @returns - the result of executing the visualize command and parsing the output
+ */
+async function runCoverageCommand(command: string, harnessName: string): Promise<any> {
+	// Get the full resolved path for the root directory of the crate
+	const directory = path.resolve(getRootDir());
+	const commmandSplit = command.split(' ');
+
+	// Get args for the command to be executed
+	const args = commmandSplit.slice(1);
+
+	const options = {
+		shell: false,
+		cwd: directory,
+	};
+
+	const globalConfig = GlobalConfig.getInstance();
+	const kaniBinaryPath = await getKaniPath('kani');
```

Comment:
We could use it with `cargo kani`, but since we render coverage using filenames, I thought it would be better to use `kani` and specify the current filename. Can easily change this or toggle. What would you advice we use here?

Reply:
> We could use it with `cargo kani`, but since we render coverage using filenames, I thought it would be better to use `kani` and specify the current filename.

I don't what this means. Both `kani` and `cargo kani` should be ready to use the coverage option and point to the right filenames when using it.

As it is right now, the extension won't be able to highlight coverage when using `cargo kani` because it'll have trouble finding other parts of the project. Let's change it so it works both ways.